### ISSUE TITLE
Speechcolor | Speechcoloroverride

### ISSUE THREAD

**Issue by soulless320** [Sphereserver/Source-X#601]
Speechcoloroverride overrides MSGs not just SAY color, this should be using the settings from 'IMSG_DEF_COLOR' or whatever msg @'color' is being used. Speech should only apply to SAY and not MSGs
![image](https://user-images.githubusercontent.com/17236327/104250138-f75f8200-543a-11eb-9d16-683a1e647ff7.png)
speechcoloroverride 01 ---> ".msg test"
![image](https://user-images.githubusercontent.com/17236327/104250755-45c15080-543c-11eb-8166-06a851704076.png)

Regular Speechcolor doesnt seem to work at all, and the below setting should never override speechcolor regardless, they should only be the default settings if no speechcolor is set.
// Default settings for Say (talkmode_say)
// Note: these settings will not override player speech settings
SAY_DEF_COLOR		946

**Comment by xwerswoodx** [comment]
Hmm, I see the reason. Not a big issue but worth to fix I think.

**Comment by soulless320** [comment]
Yes, some of these are just things I've been needing to report and haven't had time. I'm going through the list, double checking them on vanilla Sphere-X latest script pack, latest EXE and making sure they still exist. 

Gotta get them out of my todo list and make them visible :) 👍

**Comment by cbnolok** [comment]
Speechcolor is read-only on X!

**Comment by xwerswoodx** [comment]
Yes because we have speechcoloroverride instead of speechcolor, and speechcoloroverride is writeable

**Comment by soulless320** [comment]
I think its fine if speechcolor is read only, but if it is set, it should be used over SAY_DEF_COLOR
speechcoloroverride works fine for me, but it also affects MSG which is incorrect behavior

not a big bug, but nonetheless :)

**Comment by soulless320** [comment]
i'm updating the main post in this thread because after some research I was wrong on this.

**Comment by xwerswoodx** [comment]
Technically we don't need speechcolor for it, because you already have default color definitions inside messages/sphere_msgs script. I am trying to get some point to change speechcolor (possibly add new one because speechcolor holds color that sent by client if i remember correct) so what is the usage of speech color? Because if you don't want to override any item color you can just do SAY @color instead of using SPEECHCOLOROVERRIDE for item. Or maybe I am just missing something. On the other hand, if you set default colors in message file, you don't need to use override because it already override default colors but if you use @color it does change default color so it works like you want.

**Comment by soulless320** [comment]
speechcolor does hold the clients color. I'll try to explain why we need it and how it should function

the way UO was in t2a and some of renaissance every NPC had their own speech color. so when they said things it should show up in their different color. most npc tags were in a different color than their speechcolor though.  so speech color is necessary if  don't want every npc to have the same color voice (like setting in sphere_msgs). and speech color can work for "message test" but should not override "message @color test", that would be the purpose of "speechcoloroverride". if speechcolor overrides "message @color test" then all things like (tame, bonded) will show up in colors that are not correct, those should be white. 

in this way both speechcolor and speechcoloroverride will have their specific purposes, in alignment with their names. 

speechcolor : used for all "say" and "message", unless @color is provided
speechcoloroverride: used for all "say" and "message", including @color 
messages/sphere_msgs : used unless the above are not specified, its just a default

**Comment by xwerswoodx** [comment]
Uhm, okay I get it, I will look up speechcolor, because it seems like player only and I can move it under char and combine both of them but I need to check first to make sure not to damage anything.

**Comment by Soulless-1** [comment]
I don't even remember what the issue is here, closing for now.
